src/bun.js/api/JSBundler.zig

```diff
@@ -184,12 +201,11 @@ pub const JSBundler = struct {
             }
 
             if (try config.getOwnOptionalEnum(globalThis, "format", options.Format)) |format| {
-                switch (format) {
-                    .esm => {},
-                    else => {
-                        globalThis.throwInvalidArguments("Formats besides 'esm' are not implemented", .{});
-                        return error.JSError;
-                    },
+                this.format = format;
+
+                if (this.bytecode and !(this.format == .esm or this.format == .cjs)) {
+                    globalThis.throwInvalidArguments("format must be 'esm' or 'cjs' when bytecode is true", .{});
```

Comment:
`since esm doesn't really work yet.` this should not allow esm if it does not work.

Reply:
I agree, it makes sense to include esm in with this error

Reply:
It does work - it just has no impact.